integration_tests/steps/two_sided_market.py

```diff
@@ -199,3 +200,63 @@ def energy_origin(context, producer, consumer):
                 assert trade.seller_origin == producer
 
     assert trade_count > 0
+
+
+@then('trades are matched only on the Grid market')
+def trades_matched_on_grid(context):
+    # Assert that all grid trades contain Offer objects, bid trades are not tracked
+    assert all(type(trade.offer) == Offer
+               for market in context.simulation.area.past_markets
+               for trade in market.trades)
+
+    # Assert that all House 1 trades contain Bid objects
+    house1 = next(c for c in context.simulation.area.children if c.name == "House 1")
+    assert all(type(trade.offer) == Bid
+               for market in house1.past_markets
+               for trade in market.trades)
+
+    # Assert that all House 2 trades contain Offer objects
+    house2 = next(c for c in context.simulation.area.children if c.name == "House 2")
+    assert all(type(trade.offer) == Offer
+               for market in house2.past_markets
+               for trade in market.trades)
+
+
+@then('trades are matched only on the House 1 market')
```

Comment:
Some potential enhancement if area name and `Offer/Bid` is parameterised

Reply:
I thought of that too, but I thought that it would be way too confusing to understand the test afterwards. If you prefer it that way I can do it quickly, let me know

Reply:
Please leave it like is. Its not that much code and i think it is easier to understand how it is now.